### PR TITLE
Subscriptions should be version ignorant

### PR DESCRIPTION
**Pull request by Scooletz**
Connects to Particular/PlatformDevelopment#960

Backport of https://github.com/Particular/NServiceBus.Persistence.AzureStorage/pull/104

### CODE REVIEW

**Comment by Scooletz** [comment]
Ping @Particular/azure-storage-persistence-maintainers for a review.

File: src/NServiceBus.Azure.QuickTests/Subscriptions/When_subscribing.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    using System;
+    using System.Linq;
+    using Unicast.Subscriptions;
+    using NUnit.Framework;
+
+    [TestFixture]
+    [Category("AzureStoragePersistence")]
+    public class When_subscribing : BaseAzureSagaPersisterTest
+    {
+        const string Q = "address://test-queue";
```
Comment (SeanFeldman) [inline src/NServiceBus.Azure.QuickTests/Subscriptions/When_subscribing.cs]: Is VS intellisense on a strike and that's why we use one letter names?
While it's a minor, I still rather read a proper name. 😉

**Comment by Scooletz** [comment]
Ping @Particular/azure-storage-persistence-maintainers 
Please carefully review the added test and the range condition.

File: src/NServiceBus.Azure/Subscriptions/Azure/AzureSubscriptionStorage.cs [hunk 2]
```diff
@@ -76,28 +77,40 @@ void ISubscriptionStorage.Unsubscribe(Address address, IEnumerable<MessageType>
...
-                    var type = messageType;
+                    var name = messageType.TypeName;
+                    var lowerBound = name;
+                    var upperBound = GetUpperBound(name);
+
                     var query = from s in context.Subscriptions
-                                where s.PartitionKey == type.ToString() 
-                                select s;
+                        where s.PartitionKey.CompareTo(lowerBound) >= 0 &&
+                              s.PartitionKey.CompareTo(upperBound) < 0
```
Comment (yvesgoeleven) [inline src/NServiceBus.Azure/Subscriptions/Azure/AzureSubscriptionStorage.cs]: remark: there is a potential table scan here, but as this table is usually small I don't think it is an issue

**Comment by yvesgoeleven** [comment]
:+1: for me

**Pull request merged**